File: services/iposCron/model/applications/patent/Document.go
package patent

type Document struct {
	FileName      string  `json:"fileName" bson:"fileName"`
	LodgementDate string  `json:"lodgementDate" bson:"lodgementDate"`
	DocType       DocType `json:"docType" bson:"docType"`
	fileId        string  `json:"fileId" bson:"fileId"`
	Url           string  `json:"url" bson:"url"`
}

File: services/ipos/src/main/java/com/cs5224/ipos/model/documents/Document.java
package com.cs5224.ipos.model.documents;

import lombok.Data;
import org.springframework.data.mongodb.core.mapping.Field;

@Data
public class Document {

    @Field("fileName")
    String fileName;

    @Field("lodgementDate")
    String lodgementDate;

    @Field("url")
    String url; 

    @Field("docType")
    DocType docType; 

    @Data
    public static class DocType {
        @Field("description")
        private String description;

        @Field("code")
        private String code;
    }
}
